kclvm/compiler_base/3rdparty/rustc_errors/src/lib.rs

```diff
@@ -0,0 +1,293 @@
+//! Text rendering and related helper functions.
+//!
+//! Reuse 'styled_buffer.rs' in 'rustc_errors',
+//! and 'styled_buffer.rs' has been modified to fit the feature of 'Compiler-Base'.
+//!
+//! - add method `appendl()` and `putl()` to `StyledBuffer`.
+//! 
+//! - replaced the `enum Style` with generics `T: Clone + PartialEq + Eq + Style` to support extending more styles. 
+//!   `StyledBuffer` still should be valid when facing the user-defined style, rather than just supporting a built-in `enum Style`.
+//! 
+//! - add some test cases for 'StyledBuffer'.
+use termcolor::ColorSpec;
+
+pub mod styled_buffer;
+
+/// 'Style' is a trait used to specify the user customize 'XXXStyle' can be accepted by 'StyleBuffer'.
+///
+/// It provides the following method `render_style()`.
+/// render_style(&self) : render style to terminal color/font configuration.
+pub trait Style {
+    /// render style to terminal color/font configuration.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    ///    fn render_style(&self) -> ColorSpec {
+    ///         let mut spec = ColorSpec::new();
+    ///         match self{
+    ///             // For `DummyStyle::Dummy`, the font is intense and the font color is red.
+    ///             DummyStyle::Dummy => {
+    ///                 spec.set_fg(Some(Color::Red)).set_intense(true);
+    ///             }
+    ///         }
+    ///         spec
+    ///    }
+    /// ```
+    fn render_style(&self) -> ColorSpec;
```

Comment:
The name of `render_style` can be adjusted, such as `apply_spec` or `apply_color`? Because it returns a `ColorSpec` instead of a 'style'.

Reply:
rename `render_style` to `render_style_to_color_spec`.